### ISSUE TITLE
Add allow_useg to aci_epg_to_domain

### ISSUE THREAD

**Issue by codrake** [CiscoDevNet/terraform-provider-aci#82]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

An option to set the VMM Domain association to allow for micro segmentation doesn't exist in the current resource. This option is needed to be able to allow micro segmentation and currently have to set this with aci_rest resource as a workaround.

Options would be either **encap** (default) or **useg**.

### New or Affected Resource(s) + ACI Class(es):

* aci_epg_to_domain

### APIC version and APIC Platform

* V 5.0.2e on-prem

### Potential Terraform Configuration

```hcl
resource "aci_epg_to_domain" "example" {
  application_epg_dn    = "${aci_application_epg.example.id}"
  t_dn                   = "${aci_vmm_domain.example.id}"
  vmm_allow_promiscuous = "accept"
  vmm_forged_transmits  = "reject"
  vmm_mac_changes       = "accept"
  allow_useg            = "useg"
}
```

### References

* #0000

**Comment by nkatarmal-crest** [comment]
@codrake Sure i will add this param to the mentioned resource.

**Comment by nkatarmal-crest** [comment]
This issue has been fixed in the version 0.5.0 of the provider. Please try that and let me know if we can close this issue or not.
